contracts/liquidity_hub/bonding-manager/src/helpers.rs

```diff
@@ -52,85 +62,63 @@ pub fn validate_funds(deps: &DepsMut, info: &MessageInfo) -> Result<Coin, Contra
-/// global_index timestamp issues when querying the weight.
-pub fn validate_bonding_for_current_epoch(deps: &DepsMut, env: &Env) -> Result<(), ContractError> {
-    let epoch_response: EpochResponse = get_current_epoch(deps.as_ref()).unwrap();
-
-    let current_epoch = epoch_response.epoch;
-    let current_time = env.block.time.seconds();
-    const DAY_IN_SECONDS: u64 = 86_400u64;
-
-    // Check if the current time is more than a day after the epoch start time
-    // to avoid potential overflow
-    if current_epoch.id != Uint64::zero() {
-        let start_time_seconds = current_epoch
-            .start_time
-            .seconds()
-            .checked_add(DAY_IN_SECONDS);
-        match start_time_seconds {
-            Some(start_time_plus_day) => {
-                if current_time > start_time_plus_day {
-                    return Err(ContractError::NewEpochNotCreatedYet {});
-                }
-            }
-            None => return Err(ContractError::Unauthorized {}),
-        }
-    }
-
-    Ok(())
-}
-
-/// Calculates the epoch id for any given timestamp based on the genesis epoch configuration.
-pub fn calculate_epoch(
-    genesis_epoch_config: EpochConfig,
-    timestamp: Timestamp,
-) -> StdResult<Uint64> {
-    let epoch_duration: Uint64 = genesis_epoch_config.duration;
+pub fn validate_bonding_for_current_epoch(deps: &DepsMut) -> Result<(), ContractError> {
+    let config = CONFIG.load(deps.storage)?;
+    let epoch_response: EpochResponse = deps.querier.query_wasm_smart(
+        config.epoch_manager_addr.to_string(),
+        &white_whale_std::epoch_manager::epoch_manager::QueryMsg::CurrentEpoch {},
+    )?;
 
-    // if this is true, it means the epoch is before the genesis epoch. In that case return Epoch 0.
-    if Uint64::new(timestamp.nanos()) < genesis_epoch_config.genesis_epoch {
-        return Ok(Uint64::zero());
-    }
+    let reward_bucket = REWARD_BUCKETS.may_load(deps.storage, epoch_response.epoch.id)?;
 
-    let elapsed_time =
-        Uint64::new(timestamp.nanos()).checked_sub(genesis_epoch_config.genesis_epoch)?;
-    let epoch = elapsed_time
-        .checked_div(epoch_duration)?
-        .checked_add(Uint64::one())?;
+    ensure!(reward_bucket.is_some(), ContractError::EpochNotCreatedYet);
 
-    Ok(epoch)
+    Ok(())
 }
 
 // Used in FillRewards to search the funds for LP tokens and withdraw them
 // If we do get some LP tokens to withdraw they could be swapped to whale in the reply
-pub fn handle_lp_tokens(
-    info: &MessageInfo,
-    config: &white_whale_std::bonding_manager::Config,
+pub fn handle_lp_tokens_rewards(
+    deps: &DepsMut,
+    funds: &[Coin],
+    config: &Config,
     submessages: &mut Vec<SubMsg>,
 ) -> Result<(), ContractError> {
-    let lp_tokens: Vec<&Coin> = info
-        .funds
+    let lp_tokens: Vec<&Coin> = funds
         .iter()
         .filter(|coin| coin.denom.contains(".pool.") | coin.denom.contains(LP_SYMBOL))
         .collect();
```

Comment:
Well the caller could call this with wrong funds in purpose to break the contract, but this will not error. If the predicate in the filter is not met, the vector will be empty which will result on the following loop to be skipped and the funky coin will just lay in the contract forever without causing any harm.